code/functions/CMakeLists.txt

```diff
@@ -5,9 +5,16 @@ project( functions LANGUAGES CXX )
 
 # Set up the compilation environment.
 include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../SolutionTarget.cmake" )
 
 # Create the user's executable.
 add_executable( functions "Structs.h" "Structs.cpp" "functions.cpp" )
+
+# Create the "solution executable".
+add_executable( functions.sol EXCLUDE_FROM_ALL "Structs.h" "Structs.cpp" "solution/functions.sol.cpp" )
+target_include_directories(functions.sol PUBLIC ${CMAKE_SOURCE_DIR})
```

Comment:
You should instead write:

```cmake
target_include_directories(functions.sol PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
```

  - For executables `PRIVATE` is more reasonable than `PUBLIC`. (I would not have written a comment just about that though.)
  - If you use [CMAKE_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_SOURCE_DIR.html) then building this "package" as part of https://github.com/hsf-training/cpluspluscourse/blob/master/code/CMakeLists.txt would not work. [CMAKE_CURRENT_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_SOURCE_DIR.html) is designed exactly for cases like this. :wink:
  - It's a good idea to put quotation marks around any variable that describes a path. To be able to handle paths with spaces in them correctly.